xbmc/windowing/gbm/DRMAtomic.cpp

```diff
@@ -23,6 +25,57 @@
 
 using namespace KODI::WINDOWING::GBM;
 
+const std::string SETTING_VIDEOSCREEN_HW_SCALING_FILTER = "videoscreen.hwscalingfilter";
+
+enum drm_scaling_filter
+{
+  DRM_SCALING_FILTER_DEFAULT,
+  DRM_SCALING_FILTER_NEAREST_NEIGHBOR,
+};
+
+uint32_t CDRMAtomic::GetScalingFilterType(const char* type)
+{
+  if (!strcmp(type, "Nearest Neighbor"))
+    return DRM_SCALING_FILTER_NEAREST_NEIGHBOR;
+  else
+    return DRM_SCALING_FILTER_DEFAULT;
+}
+
+uint32_t CDRMAtomic::GetScalingFactor(uint32_t srcWidth,
+                                      uint32_t srcHeight,
+                                      uint32_t destWidth,
+                                      uint32_t destHeight)
+{
+  uint32_t factor_W = destWidth / srcWidth;
+  uint32_t factor_H = destHeight / srcHeight;
+  if (factor_W != factor_H)
+    return (factor_W < factor_H) ? factor_W : factor_H;
+  return factor_W;
+}
+
+bool CDRMAtomic::SetScalingFilter(struct drm_object* object, const char* name, const char* type)
+{
+  uint32_t filter_type = GetScalingFilterType(type);
+  if (SupportsPropertyAndValue(object, name, filter_type))
+    if (AddProperty(object, name, filter_type))
```

Comment:
We usually surround `if` blocks longer than a single line with `{}`. This style decision arose because indentation changes occurring far away can be confusing and harder to read.